### PR TITLE
Fix LP: 1930761 - use pre 2.9 juju vsphere templates without an arch tag

### PR DESCRIPTION
**Pull request by hmlanigan**
Juju 2.9 tries to match VM templates by series and arch.  However templates created prior to juju 2.9 do not have an arch tag.  Prefer templates which match the requested arch, but do not fail if a matching template, by series, exists.

## QA steps

``` console
sudo snap refresh juju --channel 2.8/stable
/snap/bin/juju bootstrap vsphere
/snap/bin/juju deploy ubuntu

juju upgrade-controller --build-agent
juju add-unit ubuntu
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1930761

### CODE REVIEW

File: provider/vsphere/vm_template.go [hunk 1]
```diff
@@ -138,29 +139,26 @@ func (v *vmTemplateManager) getImportedTemplate(ctx context.Context, series stri
...
 	var vmTpl *object.VirtualMachine
 	var arch string
 	if len(arches) > 0 {
 		for _, item := range seriesTemplates {
 			arch, err = v.getVMArch(ctx, item)
 			if err != nil {
-				if errors.IsNotFound(err) {
-					continue
-				}
+				logger.Debugf("failed getVMArch for template %q: %s", item.InventoryPath, err)
```
Comment (jameinel) [inline provider/vsphere/vm_template.go:150]: Do we want to ignore all errors here?
I can understand the IsNotFound but what other errors are we getting that we are essentially just turning into a Warning.

**Review by wallyworld — APPROVED** [review]

**Review by jameinel — COMMENTED** [review]
I'm happy with the general changes, I'm concerned that we're changing all errors into just a debug message that is going to be hidden by default.

File: provider/vsphere/vm_template.go (hunk 1 above)
Comment (hmlanigan) [reply provider/vsphere/vm_template.go:150]: The arch found could not be a string, or the client call to get the data failed.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**